fix: preserve generic notification data during upgrade

vikrantgravitee opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-11698
## Description

Notification data is split across two tables, but the upgrader still used a generic findAll() method that only fetched data from the parent table. Because it didn’t load the child table content, the upgrader processed incomplete entities and effectively erased the missing child data during update handling.

## Additional context

<!-- Add any other context about the PR here -->
<!-- It can be links to other PRs or docs or drawing -->
<!-- Or reproduction steps in case of bug fix -->

vikrantgravitee committed:
fix: preserve generic notification data during upgrade

phiz71 commented on gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcGenericNotificationConfigRepository.java:
```diff
@@ -257,4 +259,37 @@ public int getBatchSize() {
+        LOGGER.debug("JdbcGenericNotificationConfigRepository.findAll()");
+        try {
+            List<GenericNotificationConfig> configs = jdbcTemplate.query(
+                "SELECT id, name, notifier, config, reference_id, reference_type, use_system_proxy, created_at, organization_id, updated_at " +
```
Could you use getOrm().getSelectAll() method ?

phiz71 reviewed (commented):
Don't we have a repository test for this method?

vikrantgravitee commented:
> Don't we have a repository test for this method?

@phiz71 we did not have test file for this but I added a case.

phiz71 commented:
> > Don't we have a repository test for this method?
> 
> @phiz71 we did not have test file for this but I added a case.

I was more thinking of updating this file ;) 
https://github.com/gravitee-io/gravitee-api-management/blob/e3f23d281dfa62aab9596f82d6b307e99f77a0b6/gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/GenericNotificationConfigRepositoryTest.java#L30

vikrantgravitee commented:
> > > Don't we have a repository test for this method?
> > 
> > 
> > @phiz71 we did not have test file for this but I added a case.
> 
> I was more thinking of updating this file ;)
> 
> https://github.com/gravitee-io/gravitee-api-management/blob/e3f23d281dfa62aab9596f82d6b307e99f77a0b6/gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/GenericNotificationConfigRepositoryTest.java#L30

Thank you @phiz71, I updated test case here as well.

I did not touch it because I focused only on JDBC test cases.
